### PR TITLE
added spring boot 3.2 support

### PR DESCRIPTION
**Pull request by Donquih0te**
Support for spring boot 3.2.x version.
Support for Jetty 12.

### CODE REVIEW

**Comment by kptfh** [comment]
Great work. Will review and release it soon

File: feign-reactor-benchmarks/src/main/java/reactivefeign/benchmarks/RealRequestBenchmarks.java [hunk 1]
```diff
@@ -161,16 +166,17 @@ public rx.Observable<Void> handle(HttpServerRequest<ByteBuf> request,
...
-            public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException {
-                request.getInputStream().skip(Integer.MAX_VALUE);
-                if(target.equals(PATH_WITH_PAYLOAD)){
-                    response.addHeader("Content-Type", "application/json");
-                    response.getOutputStream().write(responseJson);
-                    response.getOutputStream().flush();
+            public boolean handle(Request request, Response response, Callback callback) throws Exception {
+                Content.Chunk chunk = request.read();
+                chunk.skip(Integer.MAX_VALUE);
+                if (request.getHttpURI().getPath().startsWith(PATH_WITH_PAYLOAD)) {
```
Comment (kptfh) [inline feign-reactor-benchmarks/src/main/java/reactivefeign/benchmarks/RealRequestBenchmarks.java]: why did you changed equals to startsWith?

File: feign-reactor-core/src/test/java/reactivefeign/DefaultMethodHandlerTest.java [hunk 2]
```diff
@@ -36,7 +37,7 @@ public void shouldCallNotDefaultMethodOnActualImplementation() throws Throwable
         DefaultMethodHandler defaultMethodHandler
                 = new DefaultMethodHandler(TestInterface.class.getMethod("defaultMethod"));
 
-        TestInterface mockImplementation = mock(TestInterface.class);
+        TestInterface mockImplementation = spy(TestInterface.class);
```
Comment (kptfh) [inline feign-reactor-core/src/test/java/reactivefeign/DefaultMethodHandlerTest.java]: Why spy?

File: feign-reactor-core/src/test/java/reactivefeign/allfeatures/AllFeaturesTest.java [hunk 3]
```diff
@@ -321,7 +332,7 @@ public void shouldReturnFirstResultBeforeSecondSent() throws InterruptedExceptio
 		AtomicInteger sentCount = new AtomicInteger();
 		AtomicInteger receivedCount = new AtomicInteger();
 
-		CompletableFuture<AllFeaturesApi.TestObject> firstReceived = new CompletableFuture<>();
+		CompletableFuture<AllFeaturesApi.TestObject> firstReceived = CompletableFuture.completedFuture(null);
```
Comment (kptfh) [inline feign-reactor-core/src/test/java/reactivefeign/allfeatures/AllFeaturesTest.java]: I don't want completed future here.
I'm going to complete it later

File: feign-reactor-core/src/test/java/reactivefeign/allfeatures/AllFeaturesTest.java [hunk 4]
```diff
@@ -411,33 +424,30 @@ public void shouldMirrorBinaryBody() {
...
 	@Test
 	public void shouldMirrorStreamingBinaryBodyReactive() throws InterruptedException {
-
-		CountDownLatch countDownLatch = new CountDownLatch(2);
-
 		AtomicInteger sentCount = new AtomicInteger();
-		ConcurrentLinkedQueue<byte[]> receivedAll = new ConcurrentLinkedQueue<>();
-
-		CompletableFuture<ByteBuffer> firstReceived = new CompletableFuture<>();
+		CompletableFuture<ByteBuffer> firstReceived = CompletableFuture.completedFuture(null);
```
Comment (kptfh) [inline feign-reactor-core/src/test/java/reactivefeign/allfeatures/AllFeaturesTest.java]: I don't want completed future here.
I'm going to complete it later

File: feign-reactor-core/src/test/java/reactivefeign/allfeatures/AllFeaturesTest.java [hunk 5]
```diff
@@ -480,7 +490,7 @@ public void shouldEncodePathParam() {
 
 	@Test
 	public void shouldEncodePathParamWithReservedChars() {
-		String PATH_PARAM = "workers?in=(\"123/321\")";
+		String PATH_PARAM = "workers?in=(\"123321\")";
```
Comment (kptfh) [inline feign-reactor-core/src/test/java/reactivefeign/allfeatures/AllFeaturesTest.java:497]: why did you change this test string?

File: feign-reactor-java11/pom.xml [hunk 6]
```diff
@@ -71,21 +71,20 @@
...
+            <artifactId>jetty-http2-server</artifactId>
             <scope>test</scope>
         </dependency>
 
         <dependency>
-            <groupId>org.eclipse.jetty</groupId>
-            <artifactId>jetty-webapp</artifactId>
-            <scope>test</scope>
+            <groupId>org.eclipse.jetty.ee10</groupId>
+            <artifactId>jetty-ee10-webapp</artifactId>
```
Comment (kptfh) [inline feign-reactor-java11/pom.xml:87]: this dependency introduces licensing issue:
Dual license: EPL-2.0, GPL-2.0-with-classpath-exception

File: feign-reactor-jetty/pom.xml [hunk 7]
```diff
@@ -84,20 +89,20 @@
         </dependency>
 
         <dependency>
-            <groupId>org.eclipse.jetty</groupId>
-            <artifactId>jetty-servlet</artifactId>
+            <groupId>org.eclipse.jetty.ee10</groupId>
+            <artifactId>jetty-ee10-servlets</artifactId>
```
Comment (kptfh) [inline feign-reactor-jetty/pom.xml:93]: this dependency introduces licensing issue:
Dual license: EPL-2.0, GPL-2.0-with-classpath-exception

File: feign-reactor-jetty/pom.xml [hunk 8]
```diff
@@ -84,20 +89,20 @@
...
             <groupId>org.eclipse.jetty.http2</groupId>
-            <artifactId>http2-server</artifactId>
+            <artifactId>jetty-http2-server</artifactId>
             <scope>test</scope>
         </dependency>
 
         <dependency>
-            <groupId>org.eclipse.jetty</groupId>
-            <artifactId>jetty-webapp</artifactId>
+            <groupId>org.eclipse.jetty.ee10</groupId>
```
Comment (kptfh) [inline feign-reactor-jetty/pom.xml:104]: this dependency introduces licensing issue:
Dual license: EPL-2.0, GPL-2.0-with-classpath-exception

File: feign-reactor-parent/pom.xml [hunk 9]
```diff
@@ -74,6 +75,59 @@
                 <scope>import</scope>
             </dependency>
 
+            <dependency>
```
Comment (kptfh) [inline feign-reactor-parent/pom.xml]: why did you add all this dependency here?
it should come from spring-boot-dependencies

File: feign-reactor-core/src/test/java/reactivefeign/allfeatures/AllFeaturesTest.java (hunk 5 above)
Comment (Donquih0te) [reply feign-reactor-core/src/test/java/reactivefeign/allfeatures/AllFeaturesTest.java:497]: jetty12 (as Servlet 6) no longer supports / as part of the path section of the URI

https://github.com/jetty/jetty.project/blob/877aaa5502d4ad9fa7dc83f3bbfc30f512222411/jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpURI.java#L1416-L1417

File: feign-reactor-java11/pom.xml (hunk 6 above)
Comment (Donquih0te) [reply feign-reactor-java11/pom.xml:87]: I have no idea what to do with this. Any suggestions? :)

File: feign-reactor-benchmarks/src/main/java/reactivefeign/benchmarks/RealRequestBenchmarks.java (hunk 1 above)
Comment (Donquih0te) [reply feign-reactor-benchmarks/src/main/java/reactivefeign/benchmarks/RealRequestBenchmarks.java]: I've reverted changes

File: feign-reactor-core/src/test/java/reactivefeign/DefaultMethodHandlerTest.java (hunk 2 above)
Comment (Donquih0te) [reply feign-reactor-core/src/test/java/reactivefeign/DefaultMethodHandlerTest.java]: I've reverted changes to use mock

File: feign-reactor-core/src/test/java/reactivefeign/allfeatures/AllFeaturesTest.java (hunk 3 above)
Comment (Donquih0te) [reply feign-reactor-core/src/test/java/reactivefeign/allfeatures/AllFeaturesTest.java]: done

File: feign-reactor-core/src/test/java/reactivefeign/allfeatures/AllFeaturesTest.java (hunk 4 above)
Comment (Donquih0te) [reply feign-reactor-core/src/test/java/reactivefeign/allfeatures/AllFeaturesTest.java]: done

File: feign-reactor-parent/pom.xml (hunk 9 above)
Comment (Donquih0te) [reply feign-reactor-parent/pom.xml]: You are right, I've reverted changes here

**Comment by aabirhassan** [comment]
Hi @kptfh @Donquih0te ,

Thank you for working on this PR to support Spring Boot 3.2.x! This feature is critical for my project, and I'm eagerly awaiting its release.

Could you please provide an estimated timeline for when this might be merged and released? It would help me plan my upgrades accordingly. Thanks in advance.

File: feign-reactor-java11/pom.xml (hunk 6 above)
Comment (kptfh) [reply feign-reactor-java11/pom.xml:87]: try to add test scope
